compiler/src/lnntoamm/Stmt.ts

```diff
@@ -0,0 +1,351 @@
+import { LPNode } from '../lp';
+import Output from './Amm';
+import Event from './Event';
+import Expr, { Ref } from './Expr';
+import opcodes from './opcodes';
+import Scope from './Scope';
+import Type from './Types';
+import { genName, TODO } from './util';
+
+export class MetaData {
+  scope: Scope
+  variables: { [name: string]: VarDef }
+  retTy: Type
+
+  constructor(
+    scope: Scope,
+    retTy: Type,
+    variables: { [name: string]: VarDef } = null,
+  ) {
+    this.scope = scope;
+    this.retTy = retTy;
+    this.variables = variables !== null ? variables : {};
+  }
+
+  get(name: string): VarDef | null {
+    if (!this.variables.hasOwnProperty(name)) {
+      return null;
+    }
+    return this.variables[name];
+  }
+
+  define(dec: VarDef) {
+    if (this.get(dec.name) !== null) {
+      throw new Error(`Can't redefine value ${dec.name}`);
+    }
+    this.variables[dec.name] = dec;
+  }
+}
+
+export default abstract class Stmt {
+  ast: LPNode
+
+  constructor(
+    ast: LPNode,
+  ) {
+    this.ast = ast;
+  }
+
+  abstract inline(amm: Output): void;
+
+  static fromAst(ast: LPNode, metadata: MetaData): Stmt[] {
+    let stmts = [];
+    if (ast.has('assignables')) {
+      let [ss, expr] = Expr.fromAssignablesAst(ast.get('assignments'), metadata);
```

Comment:
Nit: maybe don't use `ss` as a variable name anywhere? It's never the Reich name for one.